python/cuml/random_projection/rproj.pxi

```diff
@@ -0,0 +1,28 @@
+cdef extern from "random_projection/rproj_c.h" namespace "ML":
+
+    # Structure holding random projection hyperparameters
+    cdef struct paramsRPROJ:
+        int n_samples           # number of samples
+        int n_features          # number of features (original dimension)
+        int n_components        # number of components (target dimension)
+        double eps              # error tolerance according to Johnson-Lindenstrauss lemma
+        bool gaussian_method    # toggle Gaussian or Sparse random projection methods
+        double density		# ratio of non-zero component in the random projection matrix (used for sparse random projection)
+        bool dense_output       # toggle random projection's transformation as a dense or sparse matrix
+        int random_state        # seed used by random generator
+
```

Comment:
Let's expose the `johnson_lindenstrauss_min_dim` computation as well so that we can match sklearn.